public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -734,6 +734,10 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
     return getQueryHints(query.expr ?? '', result, this);
   }
 
+  getInitHints() {
+    return getInitHints(this);
```

Comment:
This seems a bit weird. Like why isn't it `this.getInitHints()`? We don't reuse it anywhere.

Reply:
We are importing this function from `./query_hints` where it is with `getQueryHints`. I think the point was to have `datasource.ts` smaller and not have all the different hints in separate file. I honestly like them together there rather than having 1 hinting function in `datasource.ts` and other in `query_hints.ts`.